packages/flutter_test/lib/src/window.dart

```diff
@@ -48,146 +46,158 @@ class TestWindow implements Window {
-    onMetricsChanged();
+    _metricsChanged();
   }
   /// Deletes any existing test physical size and returns to using the real
   /// physical size.
   void clearPhysicalSizeTestValue() {
     _physicalSizeTestValue = null;
-    onMetricsChanged();
+    _metricsChanged();
   }
 
   @override
   WindowPadding get viewInsets => _viewInsetsTestValue ??  _window.viewInsets;
-  WindowPadding _viewInsetsTestValue;
+  WindowPadding? _viewInsetsTestValue;
   /// Hides the real view insets and reports the given [viewInsetsTestValue]
   /// instead.
   set viewInsetsTestValue(WindowPadding viewInsetsTestValue) {
     _viewInsetsTestValue = viewInsetsTestValue;
-    onMetricsChanged();
+    _metricsChanged();
   }
   /// Deletes any existing test view insets and returns to using the real view
   /// insets.
   void clearViewInsetsTestValue() {
     _viewInsetsTestValue = null;
-    onMetricsChanged();
+    _metricsChanged();
   }
 
   @override
   WindowPadding get viewPadding => _viewPaddingTestValue ?? _window.padding;
-  WindowPadding _viewPaddingTestValue;
+  WindowPadding? _viewPaddingTestValue;
   /// Hides the real view padding and reports the given [paddingTestValue]
   /// instead.
   set viewPaddingTestValue(WindowPadding viewPaddingTestValue) {
     _viewPaddingTestValue = viewPaddingTestValue;
-    onMetricsChanged();
+    _metricsChanged();
   }
   /// Deletes any existing test view padding and returns to using the real
   /// viewPadding.
   void clearViewPaddingTestValue() {
     _viewPaddingTestValue = null;
-    onMetricsChanged();
+    _metricsChanged();
   }
 
   @override
   WindowPadding get padding => _paddingTestValue ?? _window.padding;
-  WindowPadding _paddingTestValue;
+  WindowPadding? _paddingTestValue;
   /// Hides the real padding and reports the given [paddingTestValue] instead.
   set paddingTestValue(WindowPadding paddingTestValue) {
     _paddingTestValue = paddingTestValue;
-    onMetricsChanged();
+    _metricsChanged();
   }
   /// Deletes any existing test padding and returns to using the real padding.
   void clearPaddingTestValue() {
     _paddingTestValue = null;
-    onMetricsChanged();
+    _metricsChanged();
   }
 
   @override
   WindowPadding get systemGestureInsets => _systemGestureInsetsTestValue ?? _window.systemGestureInsets;
-  WindowPadding _systemGestureInsetsTestValue;
+  WindowPadding? _systemGestureInsetsTestValue;
   /// Hides the real system gesture insets and reports the given [systemGestureInsetsTestValue] instead.
   set systemGestureInsetsTestValue(WindowPadding systemGestureInsetsTestValue) {
     _systemGestureInsetsTestValue = systemGestureInsetsTestValue;
-    onMetricsChanged();
+    _metricsChanged();
   }
   /// Deletes any existing test system gesture insets and returns to using the real system gesture insets.
   void clearSystemGestureInsetsTestValue() {
     _systemGestureInsetsTestValue = null;
-    onMetricsChanged();
+    _metricsChanged();
```

Comment:
Can't you just do this for all of these, and avoid adding another function?

```dart
	onMetricsChanged?.call();
```

Reply:
That's actually pretty clever. I forgot about `call()`. Changed all of them.